qiskit_experiments/base_experiment.py

```diff
@@ -99,62 +70,92 @@ def __init__(
+            The experiment data object.
         """
-        # NOTE: This method is intended to be overriden by subclasses if required.
-
         # Create new experiment data
         if experiment_data is None:
             experiment_data = self.__experiment_data__(self, backend=backend)
 
-        # Filter kwargs
-        run_options = self.__run_defaults__.copy()
-        circuit_options = {}
-        for key, value in kwargs.items():
-            if key in _TRANSPILE_OPTIONS or key in self._circuit_options:
-                circuit_options[key] = value
-            else:
-                run_options[key] = value
-
-        # Generate and run circuits
-        circuits = self.transpiled_circuits(backend, **circuit_options)
+        # Generate and transpile circuits
+        circuits = transpile(self.circuits(backend), backend, **self.transpile_options.__dict__)
+
+        # Run circuits on backend
+        run_opts = copy.copy(self.run_options)
+        run_opts.update_options(**run_options)
+        run_opts = run_opts.__dict__
+
         if isinstance(backend, LegacyBackend):
-            qobj = assemble(circuits, backend=backend, **run_options)
+            qobj = assemble(circuits, backend=backend, **run_opts)
             job = backend.run(qobj)
         else:
-            job = backend.run(circuits, **run_options)
+            job = backend.run(circuits, **run_opts)
 
         # Add Job to ExperimentData
         experiment_data.add_data(job)
 
         # Queue analysis of data for when job is finished
         if analysis and self.__analysis_class__ is not None:
-            # pylint: disable = not-callable
-            self.__analysis_class__().run(experiment_data, **kwargs)
+            self.run_analysis(experiment_data)
 
         # Return the ExperimentData future
         return experiment_data
 
+    def run_analysis(self, experiment_data, **options) -> ExperimentData:
+        """Run analysis and update ExperimentData with analysis result.
+
+        Args:
+            experiment_data (ExperimentData): the experiment data to analyze.
+            options: additional analysis options. Any values set here will
+                     override the value from :meth:`analysis_options`
+                     for the current run.
+
+        Returns:
+            The updated experiment data containing the analysis results and figures.
+
+        Raises:
+            QiskitError: if experiment_data container is not valid for analysis.
+        """
+        if self.__analysis_class__ is None:
+            raise QiskitError(f"Experiment {self._type} does not have a default Analysis class")
+
+        # Get analysis options
+        analysis_options = copy.copy(self.analysis_options)
+        analysis_options.update_options(**options)
+        analysis_options = analysis_options.__dict__
+
+        # Run analysis
+        # pylint: disable = not-callable
+        analysis = self.__analysis_class__()
```

Comment:
Why this doesn't use `self.analysis`? Seems like `self.analysis` can take some options in the constructor but there is no pass here to give that options. If I understand correctly composite analysis initializes an analysis instance without options (because we cannot hard-code these options in the class) and we should be able to initialize analysis without options. It seems like removing `**kwargs` from `self.analysis` and use `self.analysis()` here is reasonable.